File: problem07/p07.js
function isPrime(num) {
  sqrt = Math.sqrt(num);
  for (var i = 2; i <= sqrt; i++) {
    if (num % i == 0) {
      return false;
    }
  }
  return true;
}

function solution() {
  primes = [];
  lpn = 2;

  while (10001 > primes.length) {
    if (isPrime(lpn)) {
      primes.push(lpn);
    }
    lpn += 1;
  }
  return primes.slice(-1)[0]
}

console.log(solution());

File: problem07/p07.py
import math

def isPrime(num):
    sqrt = int(math.ceil(math.sqrt(num)))
    for i in range(2,sqrt + 1):
        if num % i == 0:
            return False
    return True

def solution():
    primes = []
    lpn = 2

    while 10000 > len(primes):
        if isPrime(lpn):
            primes.append(lpn)
        lpn += 1
    return primes[-1]

print solution()
